cmd/csi-provisioner/csi-provisioner.go

```diff
@@ -73,6 +73,8 @@ var (
 	metricsAddress = flag.String("metrics-address", "", "The TCP network address where the prometheus metrics endpoint will listen (example: `:8080`). The default is empty string, which means metrics endpoint is disabled.")
 	metricsPath    = flag.String("metrics-path", "/metrics", "The HTTP path where prometheus metrics will be exposed. Default is `/metrics`.")
 
+	defaultFSType = flag.String("default-fstype", "ext4", "Specify the filesystem type of the volume. If not specified,  external-provisioner will set default as `ext4`.")
```

Comment:
This would still force the driver to specify a default fstype ahead of time.  Can we just let driver determine and set this value internally?

Reply:
https://github.com/kubernetes-csi/external-provisioner/issues/328#issuecomment-528442910

Reply:
We could introduce `--default-fstype=none` as a backwards-compatible way of letting a CSI driver specify "don't set a default in external-provisioner, I'll handle that myself". "ext4" would still be the default, so nothing changes for drivers that don't use this new option.

Reply:
Yeah,  defaulting to `ext4` would be the way for now. Isnt this PR good to go then? @jsafrane @msau42 @xing-yang ?

Reply:
That's not quite what I meant. We should default to `ext4` to avoid the need for a major release, but *also* allow a CSI driver to use the modified external-provisioner without any defaulting. We can enable the latter with `--default-fstype=none`.

Reply:
Done..

Reply:
Can you add a brief explanation of what `none` means in the help message and also add a unit test for `--default-fstype=none`?

Reply:
I am actually thinking of releasing a 2.0 this cycle.  There are a number of 2.0 changes that I want to target and I think we have gathered enough that it's time to do them all.

So I think we can actually make the change to remove the defaulting altogether and not have a new argument for it.

(I do want to release a 1.6.0 first, so let's not merge this until I cut a 1.6.0)

Reply:
Removing the defaulting will be even better.

Reply:
> I am actually thinking of releasing a 2.0 this cycle. There are a number of 2.0 changes that I want to target and I think we have gathered enough that it's time to do them all.
> 
> So I think we can actually make the change to remove the defaulting altogether and not have a new argument for it.
> 
> (I do want to release a 1.6.0 first, so let's not merge this until I cut a 1.6.0)

@msau42 iiuc, the suggestion here is to remove the `defaultFSType` arg completely from this PR and keep it ready for `2.0` , Isnt it ?

Reply:
sorry to flip back and forth on this. We discussed this in the csi standup today and we may still want to provide an option to set a default so that we can do fstype source/target matching enforcement for cloning and potentially snapshots. cc @jsafrane @xing-yang @j-griffith

Reply:
@msau42 done.. nw :).. ptal .

Reply:
Do we want to wait for a 2.0 so that we can add the option without a default?

Will it be confusing if we add an option now and immediately deprecate it and change it in the next release?

Reply:
1.6 is released. And I expect this next release to be a 2.0. So let's go ahead and add the new option without a default. I have mentioned this in the deprecation notes.

Reply:
@msau42 sure . thanks!